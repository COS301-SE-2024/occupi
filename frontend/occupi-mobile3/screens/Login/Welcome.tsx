import React from 'react'
import { LinearGradient } from 'expo-linear-gradient';
import {
    Button,
    Box,
    Image,
    Center,
    Text,
    Heading,
  } from '@gluestack-ui/themed';
import { StyleSheet, View } from 'react-native';
import { router } from 'expo-router';

const GradientButton = ({ onPress, text }) => (
    <LinearGradient
    colors={['#614DC8', '#86EBCC', '#B2FC3A', '#EEF060']}
    locations={[0.02, 0.31, 0.67, 0.97]}
    start={[0, 1]}
    end={[1, 0]}
      style={styles.buttonContainer}
    >
      <Heading style={styles.buttonText} onPress={onPress}>
        {text}
      </Heading>
    </LinearGradient>
  );


const Onboarding3 = () => {
  return (
      <View style={{backgroundColor: 'white' }} p="$4">
        <Center h="$full" >
            <Image
              alt="logo"
              source={require('../../screens/Login/assets/images/Occupi/logo-white.png')}
              style={{ width: 200, height: 200 }}
              mb="$12"
              mt="$12"
            />
            <Heading alignSelf="$center" fontSize="$3xl" pl="$4" mb="$2" mt="$4">Log in. Let's Plan.</Heading>
            <Text alignSelf="$center" fontSize="$xl" p="$4" fontWeight="$light" mb="$16">
              Predict. Plan. Perfect. 
            </Text>
            <GradientButton
              onPress={() => router.push('/login')}
              text="Login"
            />
            <Text fontWeight="$bold" mt="$6" onPress={() => router.push('/signup')}>Register</Text>
        </Center>
      </View>
  )
}

const styles = StyleSheet.create({
    buttonContainer: {
      borderRadius: 15,
      marginTop: 20,
      alignSelf: 'center',
      width: 360,
      height: 50
    },
    buttonText: {
      color: 'black',
      fontSize: 16,
      textAlign: 'center',
      lineHeight: 50,
    }
  });
  
export default Onboarding3