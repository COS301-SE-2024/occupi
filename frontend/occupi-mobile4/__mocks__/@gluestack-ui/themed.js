const actualModule = jest.requireActual('@gluestack-ui/themed');

module.exports = {
  ...actualModule,
  Icon: 'Icon',
  View: 'View',
  ScrollView: 'ScrollView',
  Text: 'Text',
  Image: 'Image',
  Box: 'Box',
  // Add any other components you're using from @gluestack-ui/themed
};