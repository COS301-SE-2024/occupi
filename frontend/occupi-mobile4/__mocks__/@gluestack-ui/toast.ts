export const useToast = jest.fn().mockReturnValue({
    current: {
      show: jest.fn(),
    },
  });

  