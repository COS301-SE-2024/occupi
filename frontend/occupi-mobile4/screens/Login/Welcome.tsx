import React from 'react';
import { LinearGradient } from 'expo-linear-gradient';
import {
  Button,
  Box,
  Image,
  Center,
  Text,
  Heading,
} from '@gluestack-ui/themed';
import { StyleSheet, View } from 'react-native';
import { router } from 'expo-router';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';

const GradientButton = ({ onPress, text }) => (
  <LinearGradient
    colors={['#614DC8', '#86EBCC', '#B2FC3A', '#EEF060']}
    locations={[0.02, 0.31, 0.67, 0.97]}
    start={[0, 1]}
    end={[1, 0]}
    style={styles.buttonContainer}
  >
    <Heading style={styles.buttonText} onPress={onPress}>
      {text}
    </Heading>
  </LinearGradient>
);

const Welcome = () => {
  return (
    <View style={styles.container}>
      <Center style={styles.center}>
        <Image
          alt="logo"
          source={require('../../screens/Login/assets/images/Occupi/logo-white.png')}
          style={styles.logo}
        />
        <Heading style={styles.heading}>Log in. Let's Plan.</Heading>
        <Text style={styles.subHeading}>Predict. Plan. Perfect.</Text>
        <GradientButton
          onPress={() => router.push('/login')}
          text="Login"
        />
        <Text style={styles.registerText} onPress={() => router.push('/signup')}>Register</Text>
      </Center>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    padding: wp('4%'),
  },
  center: {
    height: '100%',
    justifyContent: 'center',
  },
  logo: {
    width: wp('50%'),
    height: wp('50%'),
    marginBottom: hp('3%'),
    marginTop: hp('3%'),
  },
  heading: {
    alignSelf: 'center',
    fontSize: wp('8%'),
    paddingLeft: wp('4%'),
    marginBottom: hp('2%'),
    marginTop: hp('4%'),
  },
  subHeading: {
    alignSelf: 'center',
    fontSize: wp('5%'),
    padding: wp('4%'),
    fontWeight: '300',
    marginBottom: hp('6%'),
  },
  buttonContainer: {
    borderRadius: 15,
    marginTop: hp('2%'),
    alignSelf: 'center',
    width: wp('90%'),
    height: hp('6%'),
  },
  buttonText: {
    color: 'black',
    fontSize: wp('4%'),
    textAlign: 'center',
    lineHeight: hp('6%'),
  },
  registerText: {
    fontWeight: 'bold',
    marginTop: hp('3%'),
    fontSize: wp('4%'),
  },
});

export default Welcome;
