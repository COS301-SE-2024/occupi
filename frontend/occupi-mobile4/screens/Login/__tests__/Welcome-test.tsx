import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { StyledProvider, Theme } from '@gluestack-ui/themed';
import Welcome from '../Welcome'; // Adjust the path to your component

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper'); // To prevent warnings about Animated module
jest.mock('expo-router', () => ({
  router: {
    replace: jest.fn(),
    push: jest.fn(),
  },
}));

const renderWithProvider = (component) => {
  return render(
    <StyledProvider theme={Theme}>
      {component}
    </StyledProvider>
  );
};

describe('Welcome component', () => {
  it('renders correctly and matches snapshot', () => {
    const tree = renderWithProvider(<Welcome />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders text correctly', () => {
    const { getByText } = renderWithProvider(<Welcome />);
    expect(getByText("Log in. Let's Plan.")).toBeTruthy();
    expect(getByText('Predict. Plan. Perfect.')).toBeTruthy();
    expect(getByText('Register')).toBeTruthy();
  });

  it('navigates to the login screen on login button press', () => {
    const { getByText } = renderWithProvider(<Welcome />);
    const loginButton = getByText('Login');

    fireEvent.press(loginButton);

    expect(require('expo-router').router.replace).toHaveBeenCalledWith('/login');
  });

  it('navigates to the signup screen on register text press', () => {
    const { getByText } = renderWithProvider(<Welcome />);
    const registerText = getByText('Register');

    fireEvent.press(registerText);

    expect(require('expo-router').router.push).toHaveBeenCalledWith('/signup');
  });
});
