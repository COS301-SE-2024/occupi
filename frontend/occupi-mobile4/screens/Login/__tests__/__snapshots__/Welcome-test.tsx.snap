// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component renders correctly and matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 30,
    }
  }
>
  <View
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    style={
      [
        undefined,
        {},
        {
          "height": "100%",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      alt="logo"
      aria-label="logo"
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      role="img"
      source={1}
      style={
        [
          undefined,
          {},
          {
            "height": 375,
            "marginBottom": 40,
            "marginTop": 40,
            "width": 375,
          },
        ]
      }
    />
    <Text
      accessibilityRole="header"
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        {
          "alignSelf": "center",
          "fontSize": 60,
          "fontWeight": "bold",
          "marginBottom": 26.5,
          "marginTop": 53.5,
          "marginVertical": 42.56,
          "paddingLeft": 30,
        }
      }
    >
      Log in. Let's Plan.
    </Text>
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          {
            "alignSelf": "center",
            "fontSize": 37.5,
            "fontWeight": "300",
            "marginBottom": 80,
            "padding": 30,
          },
        ]
      }
    >
      Predict. Plan. Perfect.
    </Text>
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          4284566984,
          4287032268,
          4289920058,
          4293849184,
        ]
      }
      endPoint={
        [
          1,
          0,
        ]
      }
      locations={
        [
          0.02,
          0.31,
          0.67,
          0.97,
        ]
      }
      startPoint={
        [
          0,
          1,
        ]
      }
      style={
        {
          "alignSelf": "center",
          "borderRadius": 15,
          "height": 80,
          "marginTop": 26.5,
          "width": 675,
        }
      }
    >
      <Text
        accessibilityRole="header"
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        onPress={[Function]}
        style={
          {
            "color": "black",
            "fontSize": 30,
            "fontWeight": "bold",
            "lineHeight": 80,
            "marginVertical": 42.56,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      onPress={[Function]}
      style={
        [
          undefined,
          {},
          {
            "fontSize": 30,
            "fontWeight": "bold",
            "marginTop": 40,
          },
        ]
      }
    >
      Register
    </Text>
  </View>
</View>
`;
